#! /usr/bin/env python
# cfg file to update velocity based position controller gains

PACKAGE = 'aerial_autonomy'

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

#       Name                         , Type     , Reconfiguration level , Description    , Default , Min  , Max
gen.add("position_gain"              , double_t , 0                     , "kp"           , 0.0     , 0.0  , 10	)
gen.add("yaw_gain"                   , double_t , 0                     , "ky"           , 0.0     , 0.0  , 10	)
gen.add("max_velocity"               , double_t , 0                     , "max_vxvyvz"   , 0.0     , 0.0  , 1   )
gen.add("max_yaw_rate"               , double_t , 0                     , "max wz"       , 0.0     , 0.0	, 1   )
gen.add("yaw_i_gain"                 , double_t , 0                     , "kiy"          , 0.0     , 0.0	, 5   )
gen.add("position_i_gain"            , double_t , 0                     , "ki"           , 0.0     , 0.0	, 5   )
gen.add("integrator_saturation_gain" , double_t , 0                     , "windinggain"  , 0.0     , 0.0	, 5   )

exit(gen.generate(PACKAGE , "Config" , "VelocityBasedPositionControllerDynamic"))
